src/proto/Harmony.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package TW.Harmony.Proto;
+option java_package = "wallet.core.jni.proto";
+
+// Input data necessary to create a signed transaction.
+message SigningInput {
+    // Chain identifier (256-bit number)
+    bytes chain_id = 1;
+
+    // Nonce (256-bit number)
+    bytes nonce = 2;
+
+    // Gas price (256-bit number)
+    bytes gas_price = 3;
+
+    // Gas limit (256-bit number)
+    bytes gas_limit = 4;
+
+    // Recipient's address.
+    string to_address = 5;
+
+    // Amount to send in wei (256-bit number)
+    bytes amount = 6;
+
+    // Optional payload
+    bytes payload = 7;
+
+    // Private key.
+    bytes private_key = 8;
+
+    // From shard ID (256-bit number)
+    bytes from_shard_id = 9;
+
+    // To Shard ID (256-bit number)
+    bytes to_shard_id = 10;
+}
+
+// Transaction signing output.
+message SigningOutput {
+    // Signed and encoded transaction bytes.
+    bytes encoded = 1;
+
+    bytes v = 2;
```

Comment:
@hewigovens if there a need for `v, r, s` or encoded would be enough?

Reply:
I think we can combine r,s,v into signature